### PR TITLE
Deep link various time elements on the chart.

### LINKED ISSUE #163: Create specific links to the different chart views
Please see https://github.com/giving-a-fuck-about-climate-change/carbondoomsday/issues/110#issuecomment-364199029.

> I want to be able link directly to the "All" graph as I find it even more fucking terrifying!
> Just to be able to send a link to people (my Dad) showing the graph is going up at a crazily consistent rate since the 1960s.

Thanks for raising this @aphillipo!

### PR DESCRIPTION
**Pull request by YPCrumble**
Fixes #163.

I'm hoping I adequately followed coding convention in this repo (which is so elegant, BTW!). Thanks for any feedback and/or suggestions!

### CODE REVIEW

**Comment by YPCrumble** [comment]
Actually, I'm seeing an issue, closing.

**Comment by YPCrumble** [comment]
The issue I faced was that I had to chain the `fetchCurrentPpms` and `queryApi` calls in `ChartInfo/index.js`. `fetchCurrentPpms` captures the `limit` parameter that the API call needs for the "All" portion of the chart. 

I used a Promise to fix the issue - happy to rework if someone has a better method.

**Comment by lwm** [comment]
Thanks @YPCrumble! Kinda beyond me on this one, hopefully someone can review ASAP.

**Comment by YPCrumble** [comment]
I also should ask - what’s the best set of URLs for the deeplinks? I’ve used /week, /month /year /all etc. But perhaps deeplinks that include keywords would be better. 

Here is a list of google’s keyword suggestions in case it inspires anyone. 

https://docs.google.com/spreadsheets/d/1Jc_QIGm3cec7D8RM4-cmO81EkQMP-j3b2Sfmv3Fq5vI

**Comment by YPCrumble** [comment]
Another thought - should probably Add a canonical tag to the base url for these deep links. Content doesn’t really change. I’ll come back to this on Monday.

**Comment by YPCrumble** [comment]
One more note on this PR (which might evolve into its own issue). Currently there are no 404's on the site at all. If you go to https://www.carbondoomsday.com/shouldnt-this-be-a-404 it shows the main page. A canonical tag is probably a decent temporary solution, which could also reduce [duplicate content](https://cl.ly/442X000S0X0m).

File: src/containers/ChartInfo/index.js [hunk 1]
```diff
@@ -188,4 +203,5 @@ ChartInfo.defaultProps = {
   fetchMonthWeekPpms: () => {},
   fetchYearPpms: () => {},
   fetchAllPpms: () => {},
+  match: null,
```
Comment (grady-lad) [inline src/containers/ChartInfo/index.js]: Should `match` be an empty object ?

**Comment by grady-lad** [comment]
@YPCrumble I think it would be better to add a 404 as a separate issue,

File: src/containers/ChartInfo/index.js [hunk 2]
```diff
@@ -42,14 +42,27 @@ const PpmResizedChart = windowSize(PpmChart);
...
   componentDidMount() {
-    // query for current ppm
-    this.props.fetchCurrentPpms();
-    // Query for five years
-    this.queryApi(this.state.rangeType);
+    // Wait for the total number of records to load
+    // before requesting ALL records from the API.
+    if (this.state.rangeType === ALL) {
+      // query for current ppm and wait for limit
+      Promise.resolve(this.props.fetchCurrentPpms()).then(() => {
```
Comment (grady-lad) [inline src/containers/ChartInfo/index.js]: `fetchCurrentPpms` already returns a promise, so I am wondering do we need the call to `fetchCurrentPpms` wrapped in a `Promise.resolve` ?

File: src/containers/ChartInfo/index.js (hunk 1 above)
Comment (grady-lad) [inline src/containers/ChartInfo/index.js]: Could this be an object ? like the following
````javascript
{
params: null
}
````
That way we guarantee that the match object will have always have a `params` prop we we don't have to check against.

**Comment by grady-lad** [comment]
@YPCrumble Just tested the pr, everything looking good for me 👍

**Comment by grady-lad** [comment]
@YPCrumble Good to hear 😄 If you have any questions about the code in the repo or just React in general, don't hesitate to drop us a message

**Pull request merged**